flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -777,8 +777,15 @@ public void dropPartition(ObjectPath tablePath, CatalogPartitionSpec partitionSp
 			// partition spec can be partial
 			List<String> partialVals = HiveReflectionUtils.getPvals(hiveShim, hiveTable.getPartitionKeys(),
 				partitionSpec.getPartitionSpec());
-			return client.listPartitionNames(tablePath.getDatabaseName(), tablePath.getObjectName(), partialVals,
-				(short) -1).stream().map(HiveCatalog::createPartitionSpec).collect(Collectors.toList());
+			if (partialVals.size() == 1 && StringUtils.isNullOrWhitespaceOnly(partialVals.get(0))) {
```

Comment:
We can simply try calling `getOrderedFullPartitionValues`, which will check if the spec is valid